AdyenActions/UI/View Controllers/DelegatedAuthentication/DAErrorViewController.swift

```diff
@@ -0,0 +1,139 @@
+//
+// Copyright (c) 2024 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+@_spi(AdyenInternal) import Adyen
+import UIKit
+
+@available(iOS 16.0, *)
+internal final class DAErrorViewController: UIViewController {
+
+    internal enum Screen {
+        case authenticationFailed(localizationParameters: LocalizationParameters?)
+        case registrationFailed(localizationParameters: LocalizationParameters?)
+        case deletionConfirmation(localizationParameters: LocalizationParameters?)
+        
+        internal var title: String {
+            switch self {
+            case let .authenticationFailed(localizationParameters):
+                localizedString(.threeds2DAApprovalErrorTitle, localizationParameters)
+            case let .registrationFailed(localizationParameters):
+                localizedString(.threeds2DARegistrationErrorTitle, localizationParameters)
+            case let .deletionConfirmation(localizationParameters):
+                localizedString(.threeds2DADeletionConfirmationTitle, localizationParameters)
+            }
+        }
+        
+        internal var image: UIImage? {
+            switch self {
+            case .authenticationFailed:
+                UIImage(named: "feedback", in: Bundle.actionsInternalResources, compatibleWith: nil)
+            case .registrationFailed:
+                UIImage(systemName: "checkmark.circle",
+                        withConfiguration: UIImage.SymbolConfiguration(weight: .ultraLight))?
+                    .withTintColor(.systemGreen,
+                                   renderingMode: .alwaysOriginal)
+            case .deletionConfirmation:
+                UIImage(systemName: "trash",
+                        withConfiguration: UIImage.SymbolConfiguration(weight: .ultraLight))?
+                    .withTintColor(.systemGray,
+                                   renderingMode: .alwaysOriginal)
+            }
+        }
+        
+        internal var message: String {
```

Comment:
A bit hard to understand from the tests but do we have tests covering all these properties to make sure we have the correct values? If not, let's have tests covering `all` these conditional return values.

Reply:
I've updated the tests to cover these cases, also renamed the test names to hopefully be more readable.